pkg/app/mssql.go

```diff
@@ -0,0 +1,366 @@
+	return err
+}
+
+func (m *MssqlDB) Insert(ctx context.Context) error {
+	log.Print("Adding entry to database")
+	insert := "/opt/mssql-tools/bin/sqlcmd -S localhost -U " + m.dbUname + " -P \"" + m.dbPass + "\" -Q " +
+		"\"USE test; INSERT INTO Inventory VALUES (1, 'banana', 150)\""
+
+	insertQuery := []string{"sh", "-c", insert}
+	_, stderr, err := m.execCommand(ctx, insertQuery)
+	if err != nil {
+		return errors.Wrapf(err, "Error while inserting data into table %s", stderr)
+	}
+	return err
+}
+
+func (m *MssqlDB) Count(ctx context.Context) (int, error) {
+	log.Print("Counting entries from database")
+	insert := "/opt/mssql-tools/bin/sqlcmd -S localhost -U " + m.dbUname + " -P \"" + m.dbPass + "\" -Q " +
+		"\"SET NOCOUNT ON; USE test; SELECT COUNT(*) FROM Inventory\" -h -1"
+
+	insertQuery := []string{"sh", "-c", insert}
+	stdout, stderr, err := m.execCommand(ctx, insertQuery)
+	if err != nil {
+		return 0, errors.Wrapf(err, "Error while inserting data into table %s", stderr)
+	}
+	rowsReturned, err := strconv.Atoi(strings.TrimSpace(strings.Split(stdout, "\n")[1]))
+	return rowsReturned, nil
+}
+
+func (m *MssqlDB) Reset(ctx context.Context) error {
+	log.Print("Reseting database")
+	delete := "/opt/mssql-tools/bin/sqlcmd -S localhost -U " + m.dbUname + " -P \"" + m.dbPass + "\" -Q " +
+		"\"DROP DATABASE test\""
+	deleteQuery := []string{"sh", "-c", delete}
+	_, stderr, err := m.execCommand(ctx, deleteQuery)
+	if err != nil {
+		return errors.Wrapf(err, "Error while inserting data into table %s", stderr)
+	}
+	return err
+}
+
+func (m *MssqlDB) Initialize(ctx context.Context) error {
+	log.Print("Initializing database")
+	createDB := "/opt/mssql-tools/bin/sqlcmd -S localhost -U " + m.dbUname + " -P \"" + m.dbPass + "\" -Q " +
+		"\"CREATE DATABASE test\""
+
+	createTable := "/opt/mssql-tools/bin/sqlcmd -S localhost -U " + m.dbUname + " -P \"" + m.dbPass + "\" -Q " +
+		"\"USE test; CREATE TABLE Inventory (id INT, name NVARCHAR(50), quantity INT)\""
+
+	execQuery := []string{"sh", "-c", createDB}
+	_, stderr, err := m.execCommand(ctx, execQuery)
+	if err != nil {
+		return errors.Wrapf(err, "Error while creating the database %s", stderr)
+	}
+
+	execQuery = []string{"sh", "-c", createTable}
+	_, stderr, err = m.execCommand(ctx, execQuery)
+	if err != nil {
+		return errors.Wrapf(err, "Error while creating table %s", stderr)
+	}
+	return err
+}
+
+func (m *MssqlDB) GetClusterScopedResources(ctx context.Context) []crv1alpha1.ObjectReference {
+	return nil
+}
+
+func (m MssqlDB) execCommand(ctx context.Context, command []string) (string, string, error) {
+	podName, containerName, err := kube.GetPodContainerFromDeployment(ctx, m.cli, m.namespace, m.deployment.Name)
+	if err != nil || podName == "" {
+		return "", "", errors.Wrapf(err, "Error  getting pod and containername %s.", m.name)
+	}
+	return kube.Exec(m.cli, m.namespace, podName, containerName, command, nil)
+}
+
+func (m *MssqlDB) getDeploymentObj() (*appsv1.Deployment, error) {
```

Comment:
let's move these utility functions to another file named something like `mssql_utils.go`